lib/fireblocks/client.go

```diff
@@ -0,0 +1,105 @@
+package fireblocks
+
+import (
+	"context"
+	"crypto/rsa"
+	"os"
+
+	"github.com/omni-network/omni/lib/apirequestor"
+	"github.com/omni-network/omni/lib/errors"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const transactionEndpoint string = "v1/transactions"
+
+type FireBlocks interface {
+	// CreateTransaction creates a new transaction on the FireBlocks API.
+	// We use raw signing by default
+	CreateTransaction(ctx context.Context, opt TransactionRequestOptions) (string, error)
+
+	// GetTransactionById retrieves a transaction by its ID.
+	GetTransactionById(ctx context.Context, transactionID string) (*TransactionResponse, error)
+
+	// CreateAndWait waits for a transaction to be signed.
+	CreateAndWait(ctx context.Context, opts TransactionRequestOptions) (*TransactionResponse, error)
```

Comment:
remember, what we want is a signature. Not sure `TransactionResponse` with `SystemMessages` is something we want?

Reply:
If this is 1:1 match of the Fireblocks API, probably, we can leave it `TransactionResponse` but then we can wrap this client in a struct called, for example, `Signer` which would expose only what we need (e.g. `Signature()` or `Sign()`) and it could behind the scenes use the Fireblocks API. Then our code would depend on a `Signer` interface and we can have the Fireblocks implementation and another one with private keys for devnet/staging.